Add all known outstanding yaml config options as of process-compose v1.7.3

daroot opened the pull request:
- per-process `description` that displays in the UI
- global and per-process `log_configuration`, including `rotation`
- global and per-process `vars` for supporting Go template expansion on configs
- `env_cmds` allows running host commands to populate env variables
- `ordered_shutdown` controls the order of process shutdown
- `is_strict` does additional checking on configuration files at startup
- `disable_env_expansion` to not propagate .env variables to processes
- `http_get.{headers,status_code}` and `working_dir` for probe commands
- `replicas` to run multiple copies of processes
- `entrypoint` alternate to `command`
- `is_elevated` for sudo/runas priviledged processes
- `extends,is_disabled,is_dotenv_disabled` for multi-file fragments and overrides.
- `launch_timeout_seconds` for daemon processes

daroot committed:
Add all known outstanding yaml config options as of process-compose v1.7.3
- per-process `description` that displays in the UI
- global and per-process `log_configuration`, including `rotation`
- global and per-process `vars` for supporting Go template expansion on configs
- `env_cmds` allows running host commands to populate env variables
- `ordered_shutdown` controls the order of process shutdown
- `is_strict` does additional checking on configuration files at startup
- `disable_env_expansion` to not propagate .env variables to processes
- `http_get.{headers,status_code}` and `working_dir` for probe commands
- `replicas` to run multiple copies of processes
- `entrypoint` alternate to `command`
- `is_elevated` for sudo/runas priviledged processes
- `extends,is_disabled,is_dotenv_disabled` for multi-file fragments and overrides.
- `launch_timeout_seconds` for daemon processes

daroot commented:
nix-ci run

adrian-gierakowski reviewed (commented):
should we also add all env vars from [here](https://github.com/F1bonacc1/process-compose/blob/d83409b3216099ee15c5825cf8413eb94c79ffd6/src/config/Flags.go#L35-L48) to here:

https://github.com/Platonic-Systems/process-compose-flake/blob/e968a94633788f5d9595d727f41c2baf0714be7b/nix/process-compose/cli.nix#L18-L27

adrian-gierakowski reviewed (commented):
there are also some missing flags [here](https://github.com/Platonic-Systems/process-compose-flake/blob/e968a94633788f5d9595d727f41c2baf0714be7b/nix/process-compose/cli.nix#L31-L73), like [disable-dotenv](https://github.com/F1bonacc1/process-compose/blob/d83409b3216099ee15c5825cf8413eb94c79ffd6/src/cmd/up.go#L35)

adrian-gierakowski commented on nix/process-compose/settings/default.nix:
```diff
@@ -82,6 +91,58 @@ in
+              '';
+            };
+
+            extends = mkOption {
+              type = types.nullOr types.path;
+              default = null;
+              example = "process-compose.base.yaml";
+              description = ''
+                Make the current configuration inherit all values in the given file.
+
+                See https://f1bonacc1.github.io/process-compose/merge#configuration-inheritance-with-extends
+              '';
+            };
```
it's nice to have typed and documented options, although it increases maintenance burden
but what's annoying is not being able to easily configure stuff which has not been explicitly added yet, so maybe we could have `extraEnvironment` for na `extraOptions` (for flags) in which people could pass arbitrary untyped config?

shivaraj-bh replied:
> maybe we could have extraEnvironment for na extraOptions (for flags) in which people could pass arbitrary untyped config? It would be similar to pattern use in nixos.

Yes. 
[freeform module](https://nixos.org/manual/nixos/stable/#sec-freeform-modules) allows that without having to define any additional options.

adrian-gierakowski replied:
yes, but if you add freeforom to a typed submodule, if mistype an existing option you will not be informed
I'd prefer to have a fully typed option separate from untyped\freeform one

srid commented:
@daroot Would you like to address these, so we can merge your PR?

daroot committed:
fix: import of vars.nix needs no extra args now

daroot committed:
new: add additional PC_ env variables

- Make disable-dotenv default to true, per @adrian-gierakowski request.
- Remove unnecessary args in vars.nix

daroot commented on nix/process-compose/settings/vars.nix:
done

shivaraj-bh committed:
Merge branch 'main' into add-options

robinanterior commented:
Did the `entrypoint` option ever make it to actually become an alternative for `command`? I see the option declaration but I don't see it ever being referenced in this PR, and when I use it, I get an eval error because it still seems to reference `command`. I can debug further but I wanted to sanity check that it's normal for there to be no other reference to `entrypoint` here than the option declaration?

shivaraj-bh commented:
> and when I use it, I get an eval error because it still seems to reference command

`entrypoint` is overridden by `command` if present. https://github.com/Platonic-Systems/process-compose-flake/blob/36678818ecbd0e06a60740a291c55b683fec79dd/nix/process-compose/settings/process.nix#L275-L278

robinanterior commented:
> > and when I use it, I get an eval error because it still seems to reference command
> 
> `entrypoint` is overridden by `command` if present.
> 
> https://github.com/Platonic-Systems/process-compose-flake/blob/36678818ecbd0e06a60740a291c55b683fec79dd/nix/process-compose/settings/process.nix#L275-L278

Was that functionality [rather than the option that declares it] introduced in a different PR? Or is that overriding handled by code outside of this codebase ?

shivaraj-bh commented:
> Was that functionality [rather than the option that declares it] introduced in a different PR? Or is that overriding handled by code outside of this codebase ?

This behaviour is defined in https://github.com/F1bonacc1/process-compose/blob/0d943455017c8eb9a9dfd2c192456df1667a17c4/src/app/process.go#L273-L279. 

(This should probably be mentioned in PC docs)

robinanterior commented:
Thanks @shivaraj-bh I didn't realize it was passed through verbatim to PC. In that case I think there's a bug somewhere, issue opening in #110.
